crates/oxc_transformer/src/typescript/namespace.rs

```diff
@@ -369,6 +377,7 @@ impl<'a> TypeScript<'a> {
                 params,
                 Some(body),
             );
+            function.scope_id.set(scope_id);
```

Comment:
Could set the correct scope flags here with:

```rs
if let Some(scope_id) = scope_id {
    function.scope_id.set(Some(scope_id));
    let is_strict = ctx.scopes().get_flags(scope_id).is_strict_mode();
    let flags = ScopeFlags::Function.with_strict_mode(is_strict);
    *ctx.scopes_mut().get_flags_mut(scope_id) = flags;
}
```

(I *think* that's correct - or are TS files always strict mode?)

Reply:
Yes, It's correct. we can use `ScopeFlags::Function | ScopeFlags::StrictMode?` directly